apache/beam
# Fix the GitHub Actions postcommit dashboard

fixes https://github.com/apache/beam/issues/26234

The intension of this PR is to fix the [GitHub Actions postcommit dashboard](http://metrics.beam.apache.org/d/80hUcTh4k/ga-post-commits-status?orgId=1) that  was added in this PR [#24176](https://github.com/apache/beam/pull/24176).

As a result, the following changes have been added:

- Fixed the dashboard's json file schema to display data
- Changed the [datasource](https://github.com/apache/beam/compare/master...akvelon:beam:26234-fix_github_actions_postcommit_dashboard?expand=1#diff-94218231238a6cf88ca8882ee3a497945acb8c796656e8e1c53741fe42ce59d2R44) to the `BeamPSQL` specified in the [beampostgresql.yaml](https://github.com/apache/beam/blob/master/.test-infra/metrics/grafana/provisioning/datasources/beampostgresql.yaml#L24) file in order to fetch data from PostgreSQL
- Changed env variable names to be consistent with the [docker-compose.yml](https://github.com/apache/beam/blob/master/.test-infra/metrics/docker-compose.yml#L85-L87) file
- The [github_actions_workflows_dashboard_sync](https://console.cloud.google.com/functions/details/us-central1/github_actions_workflows_dashboard_sync?env=gen1&project=apache-beam-testing&tab=source) function is updated to be consistent with the implementation
- The [github_actions_workflows_dashboard_sync](https://console.cloud.google.com/cloudscheduler/jobs/edit/us-central1/github_actions_workflows_dashboard_sync?project=apache-beam-testing) job scheduler is enabled

### How has it been tested?

Tested on the local machine.

1. Update the `sync_workflows.py` script by adding an entry point:
```
if __name__ == "__main__":
    githubWorkflowsGrafanaSync(None, None)
```
2. Follow the [instruction](https://cwiki.apache.org/confluence/display/BEAM/Community+Metrics#CommunityMetrics-GrafanaUI) to deploy and configure a Grafana instance locally
3. Go to the `beamsyncgithub` container: `docker exec -it "beamsyncgithub" bash`
4. Pull out data from GitHub by running the sync script `python ./sync_workflows.py`
5. Go to the `GA Post-Commits Status` Grafana dashboard and verify that data is displayed

![image](https://github.com/apache/beam/assets/38127224/b28d4b13-94d9-4662-a260-0b46dee2d5f8)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:


See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[![Build python source distribution and wheels](https://github.com/apache/beam/workflows/Build%20python%20source%20distribution%20and%20wheels/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[![Python tests](https://github.com/apache/beam/workflows/Python%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[![Java tests](https://github.com/apache/beam/workflows/Java%20Tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[![Go tests](https://github.com/apache/beam/workflows/Go%20tests/badge.svg?branch=master&event=schedule)](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.
